src/Confluent.Kafka/KafkaException.cs

```diff
@@ -23,12 +23,19 @@ namespace Confluent.Kafka
 {
     public class KafkaException : Exception
     {
-        public KafkaException(Error error, string message)
-            : base(message)
+        private Error _error;
```

Comment:
There wasn't much choice left: Error was taken by the struct type, error is the argument name, so I resorted to what was left, underscore-prefixing. Since it is a private member I think it might be okay for now.
Also, there doesn't really seem to be one definitive style guide for .NET (as previously discussed), but we should of course decide on one style for all our code (generally speaking, there must be exceptions, for exceptions.)